controllers/nodehealthcheck_controller.go

```diff
@@ -767,7 +768,33 @@ func (r *NodeHealthCheckReconciler) patchStatus(nhc, nhcOrig *remediationv1alpha
 		log.Info("Patching NHC status", "new status", nhc.Status, "patch", string(patchBytes))
 	}
 
-	return r.Client.Status().Patch(context.Background(), nhc, mergeFrom)
+	// only update lastUpdate when there were other changes
+	now := metav1.Now()
+	nhc.Status.LastUpdateTime = &now
+
+	if err := r.Client.Status().Patch(ctx, nhc, mergeFrom); err != nil {
+		return err
+	}
+
+	// Wait until the cache is updated in order to prevent reading a stale status in the next reconcile
+	// and making wrong decisions based on it. The chance to run into this is very low, because we use RequeueAfter
+	// with a minimum delay of 1 second everywhere instead of Requeue: true, but this needs to be fixed because
+	// it bypasses the controller's rate limiter!
+	err := wait.PollWithContext(ctx, 200*time.Millisecond, 5*time.Second, func(ctx context.Context) (bool, error) {
+		tmpNhc := &remediationv1alpha1.NodeHealthCheck{}
+		if err := r.Client.Get(ctx, client.ObjectKeyFromObject(nhc), tmpNhc); err != nil {
+			if apierrors.IsNotFound(err) {
+				// nothing we can do anymore
+				return true, nil
+			}
+			return false, nil
```

Comment:
Return an error?
We never return an error in [these](https://github.com/medik8s/node-healthcheck-operator/pull/245/files#diff-42b14dc2a70877933db4647434fcbe9b65a8859c7ec1911b0c9da7361c8e98b1R783-R793) lines..

Reply:
why should I return an error here? 🤔

Reply:
`func(ctx context.Context) (bool, error)` is expecting an error as the last argument that the function returns, and we haven't returned any error here. I think it would be clearer to not return an error at all and return only a bool or return at least once an error. WDYT?

Reply:
the function signature is fixed, it's an argument to wait.Poll()

Reply:
Oho, I didn't notice that it's an argument to wait.Poll()